src/utils/tracking.js

```diff
@@ -0,0 +1,59 @@
+import { init, push } from '@socialgouv/matomo-next';
+import { ssrGetCurrentEnvironment } from './environment';
+
+const env = ssrGetCurrentEnvironment();
+const MATOMO_URL = 'https://biomage.matomo.cloud/';
+// Site IDs:
+const SITE_IDS = {
+  production: 1,
+  staging: 2,
+  development: 3,
+};
+const MATOMO_SITE_ID = SITE_IDS[env];
+
+// By default tracking is only enabled on production. To test is locally
+// just manually set next variable.
+// To test a staging deployment, you'll need to go to biomage.matomo.cloud
+// and change the URL there to point to your staging env URL.
+// TODO: remove before merging
+// const trackingEnabled = env === 'production';
+const trackingEnabled = true;
+
+const initTracking = () => {
+  if (trackingEnabled === true) {
+    init({ url: MATOMO_URL, siteId: MATOMO_SITE_ID });
+  } else {
+    console.log(`tracking init disabled in env ${env}`);
+  }
+};
+
+const setTrackingId = (uid) => {
+  if (trackingEnabled === true) {
+    push(['setUserId', uid]);
+    push(['trackPageView']);
+  } else {
+    console.log(`setting tracking ID disabled in env ${env}`);
+  }
+};
+
+const resetTrackingId = () => {
+  if (trackingEnabled === true) {
+    push(['resetUserId']);
+    // we also force a new visit to be created for the pageviews after logout
+    push(['appendToTrackingUrl', 'new_visit=1']);
+  } else {
+    console.log(`setting tracking ID disabled in env ${env}`);
+  }
+};
+// the push method takes as parameter t
+const trackAnalysisLaunched = () => {
```

Comment:
I see that we have a special method to track when a user launches analysis. Why do we need a method for this, but we don't need methods for other user activities?

Reply:
we aren't tracking any specific interactions so far, only visits to the different parts of the website. I added this as an example of how to track specific events (and we agreed we do want to track when the user clicks the launch analysis). In the future we'll have to add more events (like downloading plots & tables) or other stuff but there's another ticket for that and currently our primary goal is to get tracking data and returning users.